### PR TITLE
Grafana tutorial

### PR DESCRIPTION
**Pull request by jaugsburger**

### CODE REVIEW

File: blog/2020-10-19-grafana-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+title: Build a monitoring dashboard with QuestDB and Grafana
+author: Joan
+author_title: QuestDB Team
+author_url: https://github.com/jaugsburger
+author_image_url: https://avatars.githubusercontent.com/jaugsburger
+tags: [tutorial]
+description: How to use Grafana and QuestDB to build a monitoring dashboard.
+---
+
+# Build a monitoring dashboard with QuestDB and Grafana
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:11]: remove

File: blog/2020-10-19-grafana-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+
+## Introduction
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:14]: remove the title and add `<!-- truncate -->` after the paragraph so the questdb.io/blog page has a nice summary.

File: blog/2020-10-19-grafana-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+In this tutorial you will learn how to use QuestDB as a data source for your Grafana dashboards and create visualizations using aggregate functions and sampling
+
+## What is Grafana?
+Grafana is an open-source visualization tool. It consists of a server that connects to one or more data-sources to retrieve data, which is then visualized by the user in a browser.
+
+## Grafana concepts
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:20]:
```suggestion
## Grafana's concepts
```

File: blog/2020-10-19-grafana-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+Before going ahead, let’s review 3 essential Grafana concepts that we will use in this tutorial:
+
+1. **Data source** - this is how you tell Grafana where your data is stored and how you want to access it. For the purposes of our tutorial, we will have a QuestDB server running and we will access it via Postgres Wire using the PostgreSQL data source plugin.
+2. **Dashboard** - A group of widgets that are displayed together on the same screen.
+3. **Panel** - A single visualization. Think of it as a graph/table.
+
+## Setup
+### Running Grafana
+docker run -p 3000:3000 grafana/grafana
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:29]: wrap with ```shell

File: blog/2020-10-19-grafana-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+Once the Grafana server has started, you can access it via port 3000 (http://locahost:3000).
+
+### Running QuestDB
+docker run -p 8812:8812 questdb/questdb
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:34]: wrap with ```shell

File: blog/2020-10-19-grafana-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+### Create a data source
+
+Choose PostgreSQL plugin and configure it with the following settings
+
+```
+host: localhost:8812
+database:qdb
+user: admin
+password:quest
+SSL mode:disable
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:53]: format spaces

File: blog/2020-10-19-grafana-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+alt="An image of new dashboard with a 'Add new panel' button"
+className="screenshot--shadow screenshot--docs"
+src="/img/blog/2020-10-19/add-new-panel.png"
+/>
+
+
+
+The new panel will look like this:
+<img
+alt="An blank panel after being created"
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:73]: Screenshot of a blank panel...

File: blog/2020-10-19-grafana-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+className="screenshot--shadow screenshot--docs"
+src="/img/blog/2020-10-19/blank-panel.png"
+/>
+
+Toggle the query edit box to “text edit mode” by clicking on the pencil icon. Your edit box should look like this:
+<img
+alt="How to toggle text edit mode"
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:80]: Screenshot showing how to...

File: blog/2020-10-19-grafana-tutorial.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+
+Note: Grafana does not validate that queries are read-only, therefore a Grafana user could run `drop table x` and the statement would be executed. To protect against this, you might want to run a dedicated QuestDB instance set to read-only mode. You can do this by setting `http.security.readonly=true` in your `server.conf`  <--  !!! TODO !!! replace with new doc link
+
+### Create a dashboard
+
+### Add and configure a panel
+Now that we have a data source and a dashboard, we can go ahead and add our first panel.
+
+<img
+alt="An image of new dashboard with a 'Add new panel' button"
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:64]: Screenshot of a new dashboard...

File: blog/2020-10-19-grafana-tutorial.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+### Running Grafana
+docker run -p 3000:3000 grafana/grafana
+
+Once the Grafana server has started, you can access it via port 3000 (http://locahost:3000).
+
+### Running QuestDB
+docker run -p 8812:8812 questdb/questdb
+
+### Downloading the dataset
+https://we.tl/t-4ON5T3sYmt
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:37]: I think we need to write some text for this, "You can [download](url) the dataset... blablabla"

**Review by mpsq — CHANGES_REQUESTED** [review]
Mostly minor tweaks, the article looks good ;)

File: blog/2020-10-19-grafana-tutorial.md [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+className="screenshot--shadow screenshot--docs"
+src="/img/blog/2020-10-19/panel-filtering-by-taxi-type.png"
+/>
+
+
+## Aggregations with sampling
+
+You might have noticed that we are using avg() function to calculate the average distance. The avg() function is an aggregate function, in this case, aggregating data over the specified sampling interval. This means that if the sampling interval is 1-hour, then we are in effect calculating the average distance traveled during each 1-hour interval.
+
+You can find more information on aggregate functions: here
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:143]: missing link

File: blog/2020-10-19-grafana-tutorial.md [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+```
+SELECT
+pickup_datetime AS "time",
+count()
+FROM taxi_trips
+WHERE $__timeFilter(pickup_datetime)
+SAMPLE BY $__interval;
+```
+
+This is what it looks like when viewing it over a time range of 28 days
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:165]:
```suggestion
This is what it looks like when viewing it over a time range of 28 days:
```

File: blog/2020-10-19-grafana-tutorial.md [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+
+You might have noticed that we are using avg() function to calculate the average distance. The avg() function is an aggregate function, in this case, aggregating data over the specified sampling interval. This means that if the sampling interval is 1-hour, then we are in effect calculating the average distance traveled during each 1-hour interval.
+
+You can find more information on aggregate functions: here
+
+There are also 2 key grafana functions used in the query above:
+
+`$__timeFilter(pickup_datetime)`
+This handy function tells Grafana to send the from and to time defined in the Dashboard to the QuestDB server.  Grafana translates this to:
+pickup_datetime BETWEEN '2018-02-01T00:00:00Z' AND '2018-02-28T23:59:59Z'
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:149]:
```suggestion
`pickup_datetime BETWEEN '2018-02-01T00:00:00Z' AND '2018-02-28T23:59:59Z'`.
```

File: blog/2020-10-19-grafana-tutorial.md [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+src="/img/blog/2020-10-19/panel-count-of-taxi-trips-in-a-day.png"
+/>
+
+
+
+## ASOF JOIN
+
+ASOF joins allow us to join 2 tables based on timestamp where timestamps do not exactly match.
+
+Here we are joining the taxi trips data with weather data
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:185]:
```suggestion
Here we are joining the taxi trips data with weather data:
```

File: blog/2020-10-19-grafana-tutorial.md [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+alt="A panel showing the number of trips in a day using dynamic sampling."
+className="screenshot--shadow screenshot--docs"
+src="/img/blog/2020-10-19/panel-count-of-taxi-trips-in-a-day.png"
+/>
+
+
+
+## ASOF JOIN
+
+ASOF joins allow us to join 2 tables based on timestamp where timestamps do not exactly match.
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:183]:
```suggestion
`ASOF` joins allow us to join 2 tables based on timestamp where timestamps do not exactly match.
```

File: blog/2020-10-19-grafana-tutorial.md [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+SELECT
+    pickup_datetime as "time",
+    sum(fare_amount) as sumOfFares,
+    avg(rain1H) as rain1H
+FROM (taxi_trips WHERE $__timeFilter(pickup_datetime))
+ASOF JOIN weather
+SAMPLE BY $__interval;
+```
+
+This is what it looks like for the whole month of February 2018
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:197]:
```suggestion
This is what it looks like for the whole month of February 2018:
```

File: blog/2020-10-19-grafana-tutorial.md [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+Note that the graphs above have 2 Y-axis. To enable the right Y-axis, do this, click on the yellow line next to the rainH label:
+<img
+alt="Showing how to enable 2nd Y-axis by clicking on the line next to the series name."
+className="screenshot--shadow screenshot--docs"
+src="/img/blog/2020-10-19/enabling-2nd-y-axis.png"
+/>
+
+
+In the pop-up, click on the Y-axis tab and enable use of the right axis for this series.
+
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:224]: Maybe we should write a quick conclusion/summary?

File: blog/2020-10-19-grafana-tutorial.md [hunk 18]
```diff
@@ -0,0 +1,226 @@
...
+Once the Grafana server has started, you can access it via port 3000 (http://locahost:3000).
+
+### Running QuestDB
+```shell
+docker run -p 8812:8812 questdb/questdb
+```
+
+### Downloading the dataset
+On our live [demo](http://try.questdb.io:9000/), you can find 10+ years of taxi data. For this tutorial, we have a subset of that data, the data for the whole of February 2018. You can download the compressed dataset [here](https://s3-eu-west-1.amazonaws.com/questdb.io/datasets/grafana_tutorial_dataset.tar.gz)
```
Comment (mpsq) [inline blog/2020-10-19-grafana-tutorial.md:38]:
```suggestion
On our live [demo](http://try.questdb.io:9000/), you can find 10+ years of taxi data. For this tutorial, we have a subset of that data, the data for the whole of February 2018. You can download the compressed dataset [here](https://s3-eu-west-1.amazonaws.com/questdb.io/datasets/grafana_tutorial_dataset.tar.gz).
```

**Review by mpsq — CHANGES_REQUESTED** [review]
Minor tweaks.

**Review by bluestreak01 — APPROVED** [review]

**Review by bluestreak01 — APPROVED** [review]

**Review by mpsq — APPROVED** [review]
LGTM

**Pull request merged**